src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerAttributeBindingProvider.cs

```diff
@@ -171,6 +183,18 @@ internal static TimeSpan ResolveTimeSpanFromMilliseconds(string nameOfProperty,
             return TimeSpan.FromMilliseconds(attributeValue.Value);
         }
 
+        private static async Task CreateLeaseCollectionIfNotExistsAsync(ICosmosDBService leaseCosmosDBService, string databaseName, string collectionName, int throughput)
+        {
+            try
+            {
+                await CosmosDBUtility.CreateDatabaseAndCollectionIfNotExistAsync(leaseCosmosDBService, databaseName, collectionName, null, throughput);
+            }
+            catch (DocumentClientException ex) when (ex.Message.Contains(SharedThroughputRequirementException))
+            {
+                await CosmosDBUtility.CreateDatabaseAndCollectionIfNotExistAsync(leaseCosmosDBService, databaseName, collectionName, LeaseCollectionRequiredPartitionKey, throughput);
+            }
```

Comment:
Can you explain what's happening here? We try to create without a partition key... and if we get a specific error back, we try with "/id"? In what circumstances would this be required?

Reply:
When a customer has a Database with a Shared Throughput configured, it has the requirement that all collections inside are Partitioned. If the customer selects as Database one with this setting, when we try to create the Lease collection, it will fail with that particular message. We then retry creating as partitioned. In the CFP Library it is documented that the partition key must be /id for lease collection.

Reply:
Are you sure that this message won't change? Is there a possibility to look at some error code or something instead?